src/apiUtils.js

```diff
@@ -0,0 +1,42 @@
+
+/**
+ * Register an event handler for the given event name and for the given list of layers. Valid names are: `loaded`, `updated`.
+ *
+ * The 'loaded' event will be fired when all the layers are loaded (and their 'loaded' events are fired).
+ *
+ * The 'updated' event will be fired whenever one of the layers fired an 'updated event',
+ * but throttled by requestAnimationFrame to return a maximum of one event per frame.
+ *
+ * @param {string} eventName - Supported event
+ * @param {Layer[]} layerList
+ *
+ * @memberof carto
+ * @api
+ */
+export function on (eventName, layerList, callback) {
```

Comment:
Should we show an error / warning if the user wants to listen on a layer that is not added to the map?

Reply:
Solved: we should not show the error because the layer may be added later.